### PR TITLE
Remove traitlets 4 test, add missing traitlets dependency and drop Python 3.6 support

### PR DESCRIPTION
**Pull request by martinRenou**
## References

Fix https://github.com/voila-dashboards/voila/issues/1027

## User-facing changes

None

## Backwards-incompatible changes

None

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -37,6 +37,7 @@ install_requires =
     nbclient>=0.4.0,<0.6
     nbconvert>=6.0.0,<7
     websockets>=9.0
+    traitlets>=5.0.3,<6
```
Comment (trungleduc) [inline setup.cfg:40]: By forcing `traitlets` 5, will `Voila` work with `jupyterlab` 1 and 2?

Comment (martinRenou) [reply setup.cfg:40]: You mean the labextension?

Comment (trungleduc) [reply setup.cfg:40]: I mean creating an environment with `Voila` and `JupyterLab` together. Just tested, with `traitlets>=5.0.3`, we need to bump `python_requires = >=3.7` too.

Comment (martinRenou) [reply setup.cfg:40]: > I mean creating an environment with Voila and JupyterLab together

This does not seem to be a problem (at least I don't see a problem specific to the `traitlets >=5.x` pinning):
- JupyterLab 2.x depends `jupyterlab_server >=1.1.0,<2.0.0` on https://github.com/conda-forge/jupyterlab-feedstock/blob/2.x/recipe/meta.yaml#L26
- `jupyterlab_server 1.2.0` depends on `notebook >=4.2.0` https://github.com/conda-forge/jupyterlab_server-feedstock/blob/56e32c475ac6a26562c7687ebe9b357f0cf7d071/recipe/meta.yaml#L26
- latest notebook depends on `traitlets >=4.2.1` https://github.com/conda-forge/notebook-feedstock/blob/master/recipe/meta.yaml#L46

> Just tested, with traitlets>=5.0.3, we need to bump python_requires = >=3.7 too

Thanks! Will update my PR

**Review by trungleduc — APPROVED** [review]

**Comment by trungleduc** [comment]
Thanks @martinRenou !

**Comment by martinRenou** [comment]
Thanks for the review!

**Pull request merged**